core/trino-spi/src/test/java/io/trino/spi/predicate/TestEquatableValueSet.java

```diff
@@ -295,32 +295,40 @@ public void testSubtract()
         assertEquals(EquatableValueSet.of(ID, 0L).complement().subtract(EquatableValueSet.of(ID, 0L, 1L).complement()), EquatableValueSet.of(ID, 1L));
     }
 
-    @Test(expectedExceptions = UnsupportedOperationException.class)
+    @Test
     public void testUnmodifiableCollection()
     {
-        EquatableValueSet.of(ID, 1L).getValues().clear();
+        assertThatThrownBy(() -> EquatableValueSet.of(ID, 1L).getValues().clear())
+            .isInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("cannot clear values");
     }
 
-    @Test(expectedExceptions = UnsupportedOperationException.class)
+    @Test
     public void testUnmodifiableValueEntries()
     {
-        EquatableValueSet.of(ID, 1L).getEntries().clear();
+        assertThatThrownBy(() -> EquatableValueSet.of(ID, 1L).getEntries().clear())
+            .isInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("cannot clear entries");
     }
 
-    @Test(expectedExceptions = UnsupportedOperationException.class)
+    @Test
     public void testUnmodifiableIterator()
     {
-        Iterator<Object> iterator = EquatableValueSet.of(ID, 1L).getValues().iterator();
-        iterator.next();
-        iterator.remove();
+        assertThatThrownBy(() -> Iterator<Object> iterator = EquatableValueSet.of(ID, 1L).getValues().iterator())
+            .isInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("cannot iterate values");
+        assertThatThrownBy(() -> iterator.next()).isInstanceOf(UnsupportedOperationException.class);
+        assertThatThrownBy(() -> iterator.remove()).isInstanceOf(UnsupportedOperationException.class);
```

Comment:
I guess we need to assert only remove operation.

Reply:
Hi, is this what you meant?

Reply:
That's correct, and in this case there is no point in verify the message (since there isn't any)

`() -> iterator.remove()`, can be simplified to `iterator::remove`.
Also, please wrap the line after `assertThatThrownBy`, like this:

```suggestion
        assertThatThrownBy(iterator::remove)
                .isInstanceOf(UnsupportedOperationException.class);
```